### PR TITLE
ghc: Fix Catalina Sed dependency

### PR DESCRIPTION
**Pull request by dkav**
Fix regression that broke Catalina Env.prepend_path option by not including GNU Sed. See [line 117](https://github.com/Homebrew/homebrew-core/blob/cc0bf59dd4bf2a99bf53d10383c098a7392cc201/Formula/g/ghc.rb#L117) and commit 92ddef2994d6ef3d9136f6ee3ede5594622f2487.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/g/ghc.rb
Comment (SMillerDev) [inline Formula/g/ghc.rb:50]: I'm guessing this needs to be Catalina or earlier

Comment (dkav) [reply Formula/g/ghc.rb:50]: I am not completely sure but `depends_on macos: :catalina` was added in https://github.com/Homebrew/homebrew-core/commit/a46fa896dd9d474b165a9b99aeb61eecaf3df0cf and then removed recently in https://github.com/Homebrew/homebrew-core/commit/a6f51a1bdabc74c45f294ef2061e42ada52b226f. My previous [commit](https://github.com/Homebrew/homebrew-core/commit/92ddef2994d6ef3d9136f6ee3ede5594622f2487) assumed the removed line was valid. I guess I should have re-added this line too?

Comment (dkav) [reply Formula/g/ghc.rb:51]: I assumed the line needed to re-included, so I amended the commit and re-pushed.

Comment (iMichka) [reply Formula/g/ghc.rb:51]: The thing is, we don't support Catalina anymore. So this code will never be tested on our side. This is why I removed it.

Comment (iMichka) [reply Formula/g/ghc.rb:51]: Sort of both. We could keep the `sed` accommodation maintained for a few months. Maybe drop it completely for ghc 9.8 (when we release it) and keep it in ghc 9.6? It is not a big deal.

I guess you are compiling `ghc` from source then, as we don't ship a bottle anymore? I recommend that you try to upgrade as soon as possible :)

Comment (dkav) [reply Formula/g/ghc.rb:51]: I am compiling from source. I cannot update as my machine is too old!

Comment (dkav) [reply Formula/g/ghc.rb:51]: @iMichka @SMillerDev do you need any more input from me?

I will note that are still many (100+) Catalina checks in the repository.

**Comment by Bo98** [comment]
I'm not overly convinced about the `depends_on macos: :catalina` (https://github.com/Homebrew/homebrew-core/pull/111709 indicates it would likely work fine on High Sierra), but LGTM otherwise.

**Comment by dkav** [comment]
@Bo98 I am happy to make the change. Do I need to wait for consensus before going ahead?

**Comment by Bo98** [comment]
Probably not, but I can wait until tomorrow to merge.

**Comment by dkav** [comment]
@Bo98 I made the adjustments per your comments.

**Review by iMichka — APPROVED** [review]

**Comment by iMichka** [comment]
Looks good to me :)

**Pull request merged**
